packages/core/src/html/scriptAndStyleTagProcessor.ts

```diff
@@ -7,27 +8,24 @@ const cheerio = require('cheerio');
  *
  * This is to prevent Vue compilation of script/style tags (as they are not meant to be compiled).
  *
- * @param node {Object<any, any>} node from the dom traversal
- * @param userScriptsAndStyles {array} to store scripts and style tags for hoisting
+ * @param node from the dom traversal
+ * @param userScriptsAndStyles to store scripts and style tags for hoisting
  */
-function processScriptAndStyleTag(node, userScriptsAndStyles) {
+export function processScriptAndStyleTag(node: DomElement, userScriptsAndStyles: string[]) {
   // Do not process script/style tags that are meant to be inserted in head/bottom of HTML
-  const isHeadOrBottom = node.parent.name === 'head-top'
-    || node.parent.name === 'head-bottom' || node.parent.name === 'script-bottom';
+  const isHeadOrBottom = node.parent && (node.parent.name === 'head-top'
+    || node.parent.name === 'head-bottom' || node.parent.name === 'script-bottom');
   // Do not process script/style tags that are from External
   const isExternal = userScriptsAndStyles === undefined;
   if (isHeadOrBottom || isExternal) {
     return;
   }
 
-  const idx = node.parent.children.indexOf(node);
+  const nodeSiblings = node.parent?.children ?? [node];
+  const idx = nodeSiblings.indexOf(node);
   if (idx !== -1) {
-    node.parent.children.splice(idx, 1);
+    nodeSiblings.splice(idx, 1);
   }
```

Comment:
Want to confirm with the latest update, will `idx` ever be -1?

Reply:
If somehow the node is not a child of its parent, `idx` might be -1. This should likely be guaranteed by DomHandler, and I think it's unlikely that the htmlparser will have such a case. Do you think it's safe to remove this section?

Reply:
I think it might be safe to remove because, in that strange scenario above, the resultant array of nodes may also become erroneous (to say that they are siblings of the input node).